packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      const todoRandom = {
+        ...todo,
+        id: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        ...todoUpdated1,
+        id: todoRandom.id,
+        owner: userName1,
+        privateContent: 'Private Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName1]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await user1TodoHelper.create(createResultSetName, todoRandom, setWithOwnersContent);
+          },
+        ],
+        {},
+        setWithOwnersContent,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, privateContent: null, ownersContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await user1TodoHelper.update(`update${modelName}`, todoRandomUpdated, setWithOwnerContent);
+          },
+        ],
+        {},
+        setWithOwnerContent,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, privateContent: null, ownerContent: null },
+        `onUpdate${modelName}`,
+      );
+
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe('onDelete', [], {}, setWithOwnersContent, false);
+      expect(onDeleteSubscriptionResult).toHaveLength(0);
+    });
+
+    test('Owner model auth with restricted field operations', async () => {
+      const modelName = 'TodoOwnerContentVarious';
+      const user1ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName1], schema);
+      const user2ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName2], schema);
+      const user1TodoHelper = user1ModelOperationHelpers[modelName];
+      const user2TodoHelper = user2ModelOperationHelpers[modelName];
+
+      const todo = {
+        authors: [userName1],
+        privateContent: 'Private Content',
+        ownersContent: 'Owners Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const privateResultSet = `
+        id
+        privateContent
+      `;
+      const ownerResultSet = `
+        ${privateResultSet}
+        owner
+        authors
+      `;
+      const setWithOwnerContent = `
+        ${ownerResultSet}
+        ownerContent
+      `;
+      const setWithOwnersContent = `
+        ${ownerResultSet}
+        ownersContent
+      `;
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket